package Game;

message Tile {
    required uint32 floor = 1;
    optional uint32 background = 2;
    optional uint32 foreground = 3;
    optional uint32 wall1 = 4;
    optional uint32 wall2 = 5;
    optional uint32 wall3 = 6;
    optional uint32 wall4 = 7;
    optional uint32 wall5 = 8;
    optional uint32 wall6 = 9;
    optional int32 offset1 = 10;
    optional int32 offset2 = 11;
    optional int32 offset3 = 12;
    optional int32 offset4 = 13;
};

message TileStack {
    repeated Tile tiles = 1;
}
message Map {
    required uint32 width = 1;
    required uint32 height = 2;
    required uint32 floors = 3;
    repeated TileStack tile_stacks = 4;
}

